src/core/tests/pass/serialization/serialize.cpp

```diff
@@ -14,6 +14,25 @@
 #include "openvino/util/file_util.hpp"
 #include "read_ir.hpp"
 
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/stat.h>
+#endif
+
+bool createReadOnlyFile(const std::string& filename) {
+    std::ofstream file(filename);
+    if(file.is_open()) {
+        file.close();
+#ifdef _WIN32
+        return SetFileAttributes(filename.c_str(), FILE_ATTRIBUTE_READONLY) != 0;
+#else
+        return chmod(filename.c_str(), S_IRUSR) == 0;  // Set file as read-only for owner
+#endif
+    }
+    return false;
+}
```

Comment:
CI reports compilation error here:
[2024-05-20T05:30:43.485Z] [2024-05-20 05:30:30,887] [140687447123712] build ie ERROR: ../../repos/openvino/src/core/tests/pass/serialization/serialize.cpp:23:6: error: no previous declaration for ‘bool createReadOnlyFile(const string&)’ [-Werror=missing-declarations]
That should help:
```suggestion
namespace {
bool createReadOnlyFile(const std::string& filename) {
    std::ofstream file(filename);
    if(file.is_open()) {
        file.close();
#ifdef _WIN32
        return SetFileAttributes(filename.c_str(), FILE_ATTRIBUTE_READONLY) != 0;
#else
        return chmod(filename.c_str(), S_IRUSR) == 0;  // Set file as read-only for owner
#endif
    }
    return false;
}
}
```